recipes/flatcc/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+                "verify_trace": [True, False],
+                "reflection": [True, False],
+                "native_optim": [True, False],
+                "fast_double": [True, False],
+                "ignore_const_condition": [True, False],
+    }
+    default_options = { "shared": True,
+                        "fPIC": True,
+                        "portable": False,
+                        "gnu_posix_memalign": True,
+                        "runtime_lib_only": False,
+                        "verify_assert": False,
+                        "verify_trace": False,
+                        "reflection": True,
+                        "native_optim": False,
+                        "fast_double": False,
+                        "ignore_const_condition": False
+    }
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.settings.compiler == "gcc":
+            raise ConanInvalidConfiguration("Building flatcc with MinGW is not supported")
+
+    def config_options(self):
+        if self.settings.compiler == "Visual Studio":
+            #Building flatcc shared libs with Visual Studio is broken
+            del self.options.shared
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def build(self):
+        cmake = CMake(self)
+        if self.settings.compiler == "Visual Studio":
+            cmake.definitions["BUILD_SHARED_LIBS"] = False
+        else:
+            cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
+        cmake.definitions["FLATCC_PORTABLE"] = self.options.portable
+        cmake.definitions["FLATCC_GNU_POSIX_MEMALIGN"] = self.options.gnu_posix_memalign
+        cmake.definitions["FLATCC_RTONLY"] = self.options.runtime_lib_only
+        cmake.definitions["FLATCC_INSTALL"] = True
+        cmake.definitions["FLATCC_COVERAGE"] = False
+        cmake.definitions["FLATCC_DEBUG_VERIFY"] = self.options.verify_assert
+        cmake.definitions["FLATCC_TRACE_VERIFY"] = self.options.verify_trace
+        cmake.definitions["FLATCC_REFLECTION"] = self.options.reflection
+        cmake.definitions["FLATCC_NATIVE_OPTIM"] = self.options.native_optim
+        cmake.definitions["FLATCC_FAST_DOUBLE"] = self.options.fast_double
+        cmake.definitions["FLATCC_IGNORE_CONST_COND"] = self.options.ignore_const_condition
+        cmake.definitions["FLATCC_TEST"] = False
+        cmake.configure(source_folder=os.path.join(self.source_folder, self._source_subfolder))
+        cmake.build()
+        cmake.install()
+
+    def package(self):
+        # Copy license file
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
```

Comment:
Please move the `cmake.install()` to package.
Right now, running:
`conan create . flatcc/0.6.0@` and `conan create . flatcc/0.6.0@ -kb` would not work.

Reply:
Also, please see how other recipes add caching of the cmake object.

Reply:
I changed that as well.